third_party/intel/lib/TritonIntelGPUTransforms/ScheduleLoad.cpp

```diff
@@ -0,0 +1,133 @@
+//===- ScheduleLoad.cpp ----------------------------------------------*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+/// \file
+/// This file implements a naive scheduler for loop with load/dot
+//===----------------------------------------------------------------------===//
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/Debug.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "intel/include/Dialect/TritonIntelGPU/IR/Dialect.h"
+#include "intel/include/Dialect/TritonIntelGPU/Transforms/Passes.h"
+
+#include "triton/Dialect/Triton/IR/Dialect.h"
+#include "triton/Tools/Sys/GetEnv.hpp"
+
+namespace mlir::triton::gpu::intel {
+#define GEN_PASS_DEF_TRITONINTELGPUSCHEDULELOAD
+#include "intel/include/Dialect/TritonIntelGPU/Transforms/Passes.h.inc"
+} // namespace mlir::triton::gpu::intel
+
+using namespace mlir;
+namespace tt = mlir::triton;
+namespace ttg = mlir::triton::gpu;
+namespace ttgi = mlir::triton::gpu::intel;
+
+#define DEBUG_TYPE "tritonintelgpu-schedule-load"
+
+namespace {
+
+class ScheduleLoadPass
+    : public triton::gpu::intel::impl::TritonIntelGPUScheduleLoadBase<
+          ScheduleLoadPass> {
+public:
+  // hack!!! only trace dot A/B, only back 1 level
+  SmallVector<Value> getNotVisitedUses(SmallVector<tt::DotOp> dots,
+                                       unsigned opIdx = 0) {
```

Comment:
```suggestion
                                       unsigned opIdx) {
```
This default makes the code less readable by adding extra information to remember without saving much typing